qutebrowser/app.py

```diff
@@ -837,6 +839,45 @@ def __init__(self, args):
         self.focusObjectChanged.connect(self.on_focus_object_changed)
         self.setAttribute(Qt.AA_UseHighDpiPixmaps, True)
 
+        self.new_window.connect(self._on_new_window)
+        config.instance.changed.connect(self._on_config_changed)
+
+    def _on_config_changed(self, option):
+        if option == 'tabs.undo_stack_size':
+            self._update_undo_stack_size()
+
+    def _on_new_window(self, window):
+        window.tabbed_browser.shutting_down.connect(
+            functools.partial(self._on_window_closing, window)
+        )
```

Comment:
I guess this part could stay in `Application` as it's quite generic - I guess it could emit a `window_closing(window)` signal whenever a window closes?